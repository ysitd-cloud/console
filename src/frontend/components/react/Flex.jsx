import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export default function Flex({ className, children, style }) {
  const classes = classNames('flex', className);
  return <div className={classes} style={style}>{ children }</div>;
}

Flex.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
  // eslint-disable-next-line react/forbid-prop-types
  style: PropTypes.any,
};

Flex.defaultProps = {
  className: '',
  children: null,
  style: {},
};
