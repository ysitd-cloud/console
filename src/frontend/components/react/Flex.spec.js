import React from 'react';
import renderer from 'react-test-renderer';
import Flex from './Flex';

/* globals describe, expect, it */

describe('React Flex Component', () => {
  it('match snapshot', () => {
    const component = renderer.create(<Flex className="xs12">foo</Flex>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
