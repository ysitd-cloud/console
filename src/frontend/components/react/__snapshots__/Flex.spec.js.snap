// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Flex Component match snapshot 1`] = `
<div
  className="flex xs12"
  style={Object {}}
>
  foo
</div>
`;
